### DISCUSSION TITLE
Portal render children into a div

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#463] [Q&A] [answered]
When use portal to move modal into body, it wraps the children into a div.

Could it provide an option to use no wrapper? since the element inside my modal is created dynamically, I would not expect there to be many always-exist div wrappers.

**Comment by ryansolid** [comment, marked as answer]
This isn't simply done. More information in this issue: https://github.com/solidjs/solid/pull/211

**Reply by pavulon** [reply]
So the problem is: without the container div, if the mount point is on some event's path to the top of document, it could cause an infinite loop?

**Reply by ryansolid** [reply]
More or less. It also makes it so much more of a pain to reconcile top level.

**Reply by pavulon** [reply]
I see. Would container created elsewhere work the same as the one that's created inside Portal? like eg. `<Portal container={<div/>}>`. If the container needs to be there, it might be cool to give some control over it at least. Could be useful when you use flex/grid and don't want any intermediate elements. Or maybe just enforce Portal children to always be a single node :thinking: 

I know it's too late for such API changes, I'm just thinking out loud.

**Reply by AxeemHaider** [reply]
Agree with @pavulon. Is there any way I can apply class name to that div?

**Reply by ngdangtu-vn** [reply]
I have the same problem and I also prefer @pavulon solution.

**Comment by lucianopinochet** [comment]
You can create a `body div:nth-child(2)` in a global.css
